src/app/features/location/geo.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from "@angular/core";
+import { Observable } from "rxjs";
+import { Coordinates } from "./coordinates";
+import { HttpClient } from "@angular/common/http";
+
+export interface GeoResult extends Coordinates {
+  display_name: string;
+}
+
+/**
+ * A service that uses nominatim to lookup locations {@link https://nominatim.org/}
+ * TODO maybe include email in requests {@link https://nominatim.org/release-docs/develop/api/Reverse/#other}
+ */
+@Injectable({
+  providedIn: "root",
+})
+export class GeoService {
+  readonly remoteUrl = "https://nominatim.openstreetmap.org";
+
+  constructor(private http: HttpClient) {}
+
+  /**
+   * Returns locations that match the search term
+   * @param searchTerm e.g. `Rollbergstraße Berlin`
+   * TODO countrycodes should come from the datatype?
```

Comment:
does this support multiple countrycodes? Probably should put the default into the config centrally (applying to all fields, possibly with option to overwrite)

Reply:
yes, multiple can be selected [docs](https://nominatim.org/release-docs/develop/api/Search/#result-limitation)

Reply:
set in app-wide config